tools/Customization/DevHome.Customization/Views/OptimizeDevDriveDialog.xaml.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using DevHome.Common.Extensions;
+using DevHome.Common.Services;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Serilog;
+using Windows.Storage.Pickers;
+using WinUIEx;
+
+namespace DevHome.Customization.Views;
+
+public sealed partial class OptimizeDevDriveDialog : ContentDialog
+{
+    public OptimizeDevDriveDialog(string cacheToBeMoved, string existingCacheLocation, string environmentVariableToBeSet)
+    {
+        this.InitializeComponent();
+        CacheToBeMoved = cacheToBeMoved;
+        ExistingCacheLocation = existingCacheLocation;
+        EnvironmentVariableToBeSet = environmentVariableToBeSet;
+        var stringResource = new StringResource("DevHome.Customization.pri", "DevHome.Customization/Resources");
+        OptimizeDevDriveDialogDescription = stringResource.GetLocalized("OptimizeDevDriveDialogDescription", ExistingCacheLocation, EnvironmentVariableToBeSet);
+    }
+
+    public string CacheToBeMoved

```

Comment:
As Branden mentioned elsewhere, these DPs should all just be ObservableProperties.